Sources/Sentry/SentryProfiler.mm

```diff
@@ -244,15 +208,19 @@ - (SentryEnvelopeItem *)buildEnvelopeItemForTransaction:(SentryTransaction *)tra
         profile[@"debug_meta"] = @{ @"images" : debugImages };
     }
 
+    profile[@"device_architecture"] = getCPUArchitecture();
     profile[@"device_locale"] = NSLocale.currentLocale.localeIdentifier;
     profile[@"device_manufacturer"] = @"Apple";
-    const auto model = getDeviceModel();
-    profile[@"device_model"] = model;
+    profile[@"device_model"] = getDeviceModel();
     profile[@"device_os_build_number"] = getOSBuildNumber();
-#    if TARGET_OS_IOS
-    profile[@"device_os_name"] = UIDevice.currentDevice.systemName;
-    profile[@"device_os_version"] = UIDevice.currentDevice.systemVersion;
+#    if TARGET_OS_SIMULATOR
+    const auto simulatedDevice = NSProcessInfo.processInfo.environment[@"SIMULATOR_MODEL_IDENTIFIER"];
+    if (simulatedDevice != nil) {
+        profile[@"simulated_device"] = simulatedDevice;
+    }
 #    endif
+    profile[@"device_os_name"] = getOSName();
+    profile[@"device_os_version"] = getOSVersion();
```

Comment:
Differences here: 
- we now report an OS name and version for non-ios devices
- we now report a model name for iOS simulators (so, like `iPhone14,7`) the same way as for real devices, but under the `simulated_device` key (this will go under `device` subdictionary in #2203) (@phacops I know you hate optional keys, should I just send a null value under this key for non-simulator builds?)

Reply:
I would report the model name for simulated device in the same key as real devices, but we'd have the `is_emulated` key set to `true`.

No more optional keys please :p.

Reply:
It's possible someone will care to filter results by either an M1 or Intel mac running the emulator... in this case, we wouldn't have that data. But we can always do this later if someone asks for it!

Reply:
Why wouldn't we have that data? We'd have the model name, so we can identify if it's an M1 or Intel and then if it's an emulator. I'm not sure why we'd need a separate model key when it's simulated.